File: baekjoon/stage16/1904_01타일.py
import sys

def count_binary_sequence(n):
    sequence = [1, 2]
    for i in range(3, n+1):
        sequence.append((sequence[i-2]+sequence[i-3])%15746)
    return sequence

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    sequence = count_binary_sequence(n)
    print(sequence[n-1])

File: problem-solving/baekjoon/1904_01타일.go
package main

import (
	"fmt"
	"os"
	"bufio"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n int
	fmt.Fscanln(reader, &n)
	sequence := countBinarySequence(n)
	fmt.Println(sequence[n-1])
}

func countBinarySequence(n int) (sequence []int) {
	sequence = append(sequence, 1, 2)
	for i := 3; i<n+1; i++ {
		sequence = append(sequence, (sequence[i-2]+sequence[i-3])%15746)
	}
	return
}